Sources/NIOCore/NIOSendable.swift

```diff
@@ -17,3 +17,9 @@ public typealias NIOSendable = Swift.Sendable
 #else
 public typealias NIOSendable = Any
 #endif
+
+#if swift(>=5.6)
+@preconcurrency public protocol NIOPreconcurrencySendable: Sendable {}
+#else
+public protocol NIOPreconcurrencySendable {}
```

Comment:
I still believe these should be non `public`. They are just marker types for internal convenience. Adopters, still we see the correct types as `Sendable` and we are not exposing unnecessary protocols publicly.
```suggestion
@preconcurrency protocol NIOPreconcurrencySendable: Sendable {}
#else
public NIOPreconcurrencySendable {}
```

Reply:
Given that we've already shipped NIOSendable I think the symmetry argument shouldn't be overlooked here.

Reply:
I get that point, but should we expose more public API that is just an implementation detail because of symmetry reasons?

Reply:
It needs to be public, otherwise public protocols (e.g. `ChannelOption` or `EventLoop`) can **not** inherit from it.